### PR TITLE
Big redis refactor part 1

### PR DESCRIPTION
**Pull request by thesecretmaster**
Refactored a decent amount of the redis stuff to be less brittle. Not fully done, but ran out of time and haven't tested everything. It'd be great if someone could look over this and merge it, otherwise I'll prolly finish+test this next weekend.

### CODE REVIEW

File: app/models/redis/site_setting.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Redis::SiteSetting
+  PREFIX = "site_setting"
+
+  class << self
```
Comment (ArtOfCode-) [inline app/models/redis/site_setting.rb]: Prefer `def self.bool` style for defining static methods for consistency with the rest of metasmoke

File: config/environments/development.rb [hunk 2]
```diff
@@ -65,8 +65,6 @@
     s3_region: 'us-east-2'
   }
 
-  Aws.config.update({
-    region: 'us-east-1',
-    credentials: Aws::Credentials.new(AppConfig["aws"]["access_token"], AppConfig["aws"]["secret_token"])
-  })
+  Aws.config.update(region: 'us-east-1',
```
Comment (ArtOfCode-) [inline config/environments/development.rb]: AWS is an acronym, can we add it to inflector config and make it `AWS`?

Comment (Undo1) [reply config/environments/development.rb]: That's how [aws/aws-sdk-ruby](https://github.com/aws/aws-sdk-ruby) named everything. Kinda inclined to leave it that way, even if it looks weird.

Comment (ArtOfCode-) [reply config/environments/development.rb]: :+1: :(

File: app/models/redis/site_setting.rb (hunk 1 above)
Comment (thesecretmaster) [reply app/models/redis/site_setting.rb]: Ohhhh, I thought you were commenting on line 2, not line 4. I really prefer the class << self notation, but I can change it.

**Pull request merged**